epub33/core/index.html

```diff
@@ -1560,6 +1557,18 @@ <h6>The <code>metadata</code> Element</h6>
 
 						</section>
 
+						<section id="sec-package-elem-value">
+							<h5>Metadata Values</h5>
+
+							<p>When a child element of the <a href="#sec-metadata-elem"><code>metadata</code>
+									element</a> has mandatory text content, that content is referred to as the value of
+								the element in the explanatory descriptions.</p>
```

Comment:
I think what you're proposing might make the text more confusing. The element definitions talk in places about their "value". The meta element has a value, for example, but to get to a "metadata expression" you need to also be talking about what is expressed in its property attribute (or the inherent semantics of the DCMES elements).

Looking again at this though, what might help is to be even more specific. We know exactly what elements this applies to - the DCMES elements and meta - so we can list those.

I'm fine with setting value up as a definition that we can link from, though. That makes sense.

> Finally, it would perhaps be easier to define the "value" in the general case (empty or not)

Note completely sure what you mean about defining the "value"? The elements already have a content model of text in their definitions. Are you suggesting we try to further define what that means?

Reply:
> I think what you're proposing might make the text more confusing. The element definitions talk in places about their "value". The meta element has a value, for example, but to get to a "metadata expression" you need to also be talking about what is expressed in its property attribute (or the inherent semantics of the DCMES elements).

OK fair enough. I'm not too hung-up on _not_ using "element value", as long as the term is a proper definition that can be referenced. In fact, I came here from the RS spec preview, when I found "element value" and was confused by that terminology which didn't sound familiar (from HTML or XML). So it's probably OK if uses of the term can be linked to the definition.

> Note completely sure what you mean about defining the "value"? The elements already have a content model of text in their definitions. Are you suggesting we try to further define what that means?

I meant that the "value" of an element is not defined by XML, so we should be more explicit about what we mean here. DOM defines a [`nodeValue`](https://dom.spec.whatwg.org/#dom-node-nodevalue) attribute for nodes, which for elements is unambiguously defined as the [descendant text context](https://dom.spec.whatwg.org/#concept-descendant-text-content). I think we should use a similar unambiguous definition of what is exactly meant by "value" here.

The current text says:
> "When a child element of the metadata element has mandatory text content, that content is referred to as the value of the element in the explanatory descriptions."

As all the elements require a non-empty value, why make a case ("when") for mandatory text content when defining what we refer to when we say "the value of the element"?

What I suggest is something along these lines:

- add a definition for "the value of the element" (not my favorite term, but OK), based on a concept similar to DOM's _descendant text content_, without the "when".
- add the MUST not be empty statement for that term just defined, based on INFRA.

So a lot like what is already there, but more explicit :)

Reply:
> I'm not too hung-up on _not_ using "element value"

For context, this complaint came a couple of revisions back that we were using "value" for element text content when the XML spec doesn't define such a concept. The prime reason I want to stick with it is that changing to something else like "metadata expression" means we probably also have to pick through the whole spec and try to rewrite the references so that they also make sense about the text content being the expression. This made people happy at the time, so I'd rather not try to reinvent it, not that I necessarily love the terminology either. 😄 

> * add a definition for "the value of the element" (not my favorite term, but OK), based on a concept similar to DOM's _descendant text content_, without the "when".

That's probably too complex as EPUB doesn't have element nesting. Even the DOM definition of child text content is confusing in our context, as there's only one instance of child text content so nothing to concatenate.

An element has text content and that's it. I don't know what else we can compare that with, but I don't have time today to skim the usual specs for definitions.

I did make a few changes before signing off last night, like ditching the "when", adding a specific list of elements this applies to, and changing the wording to non-empty. If you have any thoughts on what we can do with text content, though, let me know.

Reply:
> An element has text content and that's it.

I wasn't thinking you could put anything but text in an element, but I suppose comments, processing instructions and even cdata blocks are all valid. Epubcheck isn't complaining when I add them.

Given that, the child text content definition is correct since there could be a concatenation in some extreme cases.

Reply:
I've made some more changes to add the child text content definition and infra whitespace handling to the core spec, and also removed the exception for meta properties defining their own whitespace handling from the RS. Let me know if you're happier with the sections now (I'm going to close these issues just because they reference out of date prose now).

Reply:
> Let me know if you're happier with the sections now

Looks better to me now! Thanks for the changes (and for bearing with my nitpicking 😅).